### PR TITLE
added a new experimental game_type

### PR DESCRIPTION
**Pull request by tstocker79**

### CODE REVIEW

File: src/models/animal.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
+export interface AnimalConstructor {
+    new (name: string, creator: string, code: string): Animal;
+}
+
+export class Animal {
+    name!: string;
```
Comment (OtherwiseJunk) [inline src/models/animal.ts:6]: what's the ! after the name do?

File: src/services/animalGameServiceFactory.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { GameType } from '../models/game-type';
+import { LoneAntAnimalGameService } from './loneAntAnimalGameService';
+import { SpawningAntsAnimalGameService } from './spawningAntsAnimalGameService';
+
+var loneAntInstance: LoneAntAnimalGameService | undefined;
+var spawningAntsInstance: SpawningAntsAnimalGameService | undefined;
+
+export class AnimalGameServiceFactory {
+  static CreateAnimalGameService(gameType: GameType) {
+    if (gameType === GameType.LoneAnt) {
```
Comment (OtherwiseJunk) [inline src/services/animalGameServiceFactory.ts:10]: This should probably be a switch, right?

File: src/models/animal.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    column!: number;
+    row!: number;
+    score!: number;
+    error: string | undefined;
+    color!: string;
+    creator!: string;
+    doStep!: Function;
+
+    public static CreateAnimal(Ctor: AnimalConstructor, antName: string, creator: string, code: string): Animal {
+      return new Ctor(antName, creator, code);
```
Comment (OtherwiseJunk) [inline src/models/animal.ts:16]: probably just being a derp and missed it, but what is a CTOR?

File: src/services/loneAntAnimalGameService.ts [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+import { AnimalAction } from '../models/animal-action';
+import { Animal } from '../models/animal';
+import { Board } from '../models/board';
+import { Game } from '../models/game';
+import { Food } from '../models/food';
+
+const FOOD_PERCENTAGE = +(process.env.FOOD_PERCENTAGE || 0.05);
+const loneAntsDirectory = './ants';
+const MAX_ANTS = +(process.env.MAX_ANTS || 10);
+const GRID_WIDTH = +(process.env.GRID_WIDTH || 200);
```
Comment (OtherwiseJunk) [inline src/services/loneAntAnimalGameService.ts:12]: Should the common config elements like grid width/height, ticks per sec, etc that would be the same be pulled before and passed into the game service start method or w/e?

File: src/services/loneAntAnimalGameService.ts [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+
+  generateGrid(height: number, width: number) {
+    const newGrid = Array(height).fill([]);
+    for (var row = 0; row < height; row++) {
+      newGrid[row] = Array(width).fill(1);
+    }
+    return newGrid;
+  }
+
+  generateFood(height: number, width: number, foodPercentage: number) {
```
Comment (OtherwiseJunk) [inline src/services/loneAntAnimalGameService.ts:107]: Does it make sense to have generateFood take the height/width/foodPercentage as an argument when they're consts on the class?

File: src/models/animal.ts (hunk 3 above)
Comment (tstocker79) [reply src/models/animal.ts:16]: constructor, it's a typescript thing that I have not fully digested yet.

File: src/services/loneAntAnimalGameService.ts (hunk 4 above)
Comment (tstocker79) [reply src/services/loneAntAnimalGameService.ts:12]: these should be separate and not really shared. These games will diverge wildly in the future.

File: src/services/loneAntAnimalGameService.ts (hunk 5 above)
Comment (tstocker79) [reply src/services/loneAntAnimalGameService.ts:107]: the constants won't be solely used for long, custom games with different sizes / ticks will be sent in with some new endpoints that are coming

**Pull request merged**